### PR TITLE
[AIRFLOW-2783] Implement eslint to check Javascript code quality

### PR DESCRIPTION
**Pull request by verdan**
Make sure you have checked _all_ steps below.

### JIRA
- [x] My PR addresses the following [Airflow 2783](https://issues.apache.org/jira/browse/AIRFLOW-2783) issues and references them in the PR title. 

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:
This PR implements the `eslint` to verify (and fix) the Javascript code errors/conventions. For now it only implements the Airbnb's JS config, without any alteration, and can be improved based on community's feedback. 

One can run the following two commands to run eslint. 
```javascript
npm run lint // This will find all the issues in the javascript code

npm run lint:fix // This will find all the issues in the javascript code, and try to fix them all
```

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
No tests needed for this chage. 

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### Documentation
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality
- [x] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by verdan** [comment]
Build failed because of Flaky test case. 
Travis (commit) build result of local repository: https://travis-ci.org/verdan/incubator-airflow/builds/407980551

**Comment by Fokko** [comment]
We should fix the flaky test soon. Can you rebase onto master?

**Comment by Fokko** [comment]
Travis should also run the linter, right?

**Comment by verdan** [comment]
@Fokko with the current configuration of `eslint` there are 1056 problems in existing code (it includes all the jinja template tags in JS code which eslint marks as error). 
I'll create a separate PR to fix all the existing issues, and then we'd be able to make travis run the linter and also run linter before compiling the build automatically.

**Comment by Fokko** [comment]
Nice. We can also tell Travis to allow to build to fail: https://docs.travis-ci.com/user/customizing-the-build/#Rows-that-are-Allowed-to-Fail

File: airflow/www_rbac/.eslintignore [hunk 1]
```diff
@@ -0,0 +1,10 @@
+**/*{.,-}min.js
+**/*.sh
+**/*.py
+gantt-chart-d3v2.js
+jqClock.min.js
+coverage/**
+static/dist/*
+static/docs/*
+node_modules/*
+node_modules*/*
```
Comment (tedmiston) [inline airflow/www_rbac/.eslintignore]: What's the second node_modules entry for?

**Comment by tedmiston** [comment]
This looks awesome.  Do you think it makes sense to add the 2 npm commands into the docs somewhere... maybe in CONTRIBUTING.md like how we mention linting for Python code?

**Comment by tedmiston** [comment]
By the way, I have fixed the `test_mark_success_no_kill` build error in https://github.com/apache/incubator-airflow/pull/3642.

File: airflow/www_rbac/.eslintignore (hunk 1 above)
Comment (verdan) [reply airflow/www_rbac/.eslintignore]: nice catch @tedmiston . In fact, we don't have to mention the `node_modules` at all here in .eslintignore file. ESLint always ignores files in `/node_modules/*` and `/bower_components/*` by default. Will remove that.

**Comment by verdan** [comment]
@tedmiston that's actually a good point. I've added the JS linting documentation in CONTRIBUTING.md file. 

Travis Build Result: https://travis-ci.org/verdan/incubator-airflow/builds/408402612

Followup Jira issue: https://issues.apache.org/jira/browse/AIRFLOW-2803 (To fix all the linting issues)

**Review by bolkedebruin — APPROVED** [review]

**Pull request closed without merging**